### PR TITLE
Added: UseMemo, UseEffect, context 

### PR DESCRIPTION
**Pull request by Michael-Ananev**

### CODE REVIEW

File: src/todo/components/TodoItem.tsx [hunk 1]
```diff
@@ -46,7 +48,8 @@ const TodoItem: React.FC<TodoItemProps> = ({ todo, index, setCheckedToDo, remove
     }
 
     return (
-        <>     
+        <>
```
Comment (dmitrymokienko) [inline src/todo/components/TodoItem.tsx]: You can remove React Fragment wrapper due to the fact you have  the div one

File: src/todo/components/TodoList.tsx [hunk 2]
```diff
@@ -1,16 +1,22 @@
...
 const ToDo = () => {
-    const [name, setName] = useState('');
+    
     const [todos, setTodos] = useState(DEFAULT_TODO_ITEM);
     const [title, setTitle] = useState('')
     const [description, setDescription] = useState('')
 
-    const handleAddTask = e => {
+
+    useEffect(() => {
```
Comment (dmitrymokienko) [inline src/todo/components/TodoList.tsx]: IMHO it's not good solution
better way to add/delete/update items in the appropriate handlers

File: src/todo/components/TodoList.tsx [hunk 3]
```diff
@@ -34,6 +40,36 @@ const ToDo = () => {
...
+        setTodos(prevTodos => 
+            prevTodos.map(todo => ({...todo, isChecked: true})))
+    }
+
+    useEffect(() => {
+        const storedTodos = localStorage.getItem("todos")
+        if(storedTodos) {
+            setTodos(JSON.parse(storedTodos))
+        }
+    }, [])
```
Comment (dmitrymokienko) [inline src/todo/components/TodoList.tsx]: It's ok

File: src/todo/components/TodoList.tsx (hunk 2 above)
Comment (dmitrymokienko) [reply src/todo/components/TodoList.tsx]: as for me, better order code by the rules

first - hooks: useState, useMemo, useEffect
then handlers
then variables for render 
and at the end - render functions

so code will be easy readable

Comment (Michael-Ananev) [reply src/todo/components/TodoList.tsx]: Do you mean just swap places?

File: src/todo/components/TodoList.tsx (hunk 3 above)
Comment (Michael-Ananev) [reply src/todo/components/TodoList.tsx]: see

File: src/todo/components/TodoItem.tsx (hunk 1 above)
Comment (Michael-Ananev) [reply src/todo/components/TodoItem.tsx]: Ok, just deleted the main div with class todo-item, the rest is left

File: src/todo/components/TodoList.tsx (hunk 2 above)
Comment (dmitrymokienko) [reply src/todo/components/TodoList.tsx]: yes

Comment (Michael-Ananev) [reply src/todo/components/TodoList.tsx]: Can I leave "memoizedTodoItem" in its place? I moved the "useEffect" hooks to the top. Just "memoizedTodoItem it's already render?

Comment (dmitrymokienko) [reply src/todo/components/TodoList.tsx]: it's up to you

File: src/todo/components/TodoList.tsx [hunk 4]
```diff
@@ -52,19 +98,28 @@ const ToDo = () => {
                     onChange={(e) => setDescription(e.target.value)}
                     placeholder="Enter task description"
                 />
-                </div>
+                <button type="submit">Add task</button>
+                </form>
+                <button onClick={removeAllTodos}>Remove All todos</button>
+                <button onClick={markAllTodosAsChecked}>Mark All Todos As Checked</button>
+                <button onClick={async()=>{
```
Comment (dmitrymokienko) [inline src/todo/components/TodoList.tsx]: better to move it to the handler func

Comment (Michael-Ananev) [reply src/todo/components/TodoList.tsx]: I don’t understand how to move them to the handle function?))

**Pull request merged**